Build errors on AIX and IBM i

```
In file included from unix/getmode.cc:10:0:
./bsd/unistd.h:66:5: error: conflicting declaration of 'int getpeereid(int, uid_t*, gid_t*)' with 'C' linkage
 int getpeereid(int s, uid_t *euid, gid_t *egid);
     ^~~~~~~~~~
In file included from ./libuv-1.38.1/include/uv/unix.h:30:0,
                 from ./libuv-1.38.1/include/uv.h:66,
                 from ./utils.h:3,
                 from ./getmode.h:7,
                 from unix/getmode.cc:1:
/QOpenSys/pkgs/lib/gcc/powerpc-ibm-aix6.1.0.0/6.3.0/include-fixed-7.1/sys/socket.h:132:9: note: previous declaration with 'C++' linkage
 int     getpeereid(int, uid_t *__restrict__, gid_t *__restrict__);
         ^~~~~~~~~~
make: *** [/QOpenSys/pkgs/lib/R/etc/Makeconf:166: unix/getmode.o] Error 1
```

This is due to AIX header files declaring the uid and gid pointers as `restrict` pointers and it's included in a C++ source file. The getpeereid function appears to be unused in this code so its definition can be removed or commented out.

In addition, Makevars does not build the necessary BSD compatibility shims, so there are missing symbol errors at runtime.

This is similar to #278, but the symbol error is not on libuv.

--- a/src/Makevars
+++ b/src/Makevars
@@ -17,7 +17,7 @@ endif
 ifeq ($(UNAME), OpenBSD)
 PKG_LIBS += -lkvm
 endif
-ifeq ($(UNAME), Linux)
+ifneq ($(filter $(UNAME), Linux AIX OS400),)
 OBJECTS +=  bsd/setmode.o bsd/strmode.o bsd/reallocarray.o
 endif
 

--- a/src/bsd/unistd.h
+++ b/src/bsd/unistd.h
@@ -63,7 +63,6 @@ void closefrom(int lowfd);
 void setproctitle_init(int argc, char *argv[], char *envp[]);
 void setproctitle(const char *fmt, ...);
 
-int getpeereid(int s, uid_t *euid, gid_t *egid);
 __END_DECLS
 
 #endif

